src/build_log.cc

```diff
@@ -151,10 +151,20 @@ bool BuildLog::RecordCommand(Edge* edge, int start_time, int end_time,
     Entries::iterator i = entries_.find(path);
     LogEntry* log_entry;
     if (i != entries_.end()) {
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+      log_entry = i->second.get();
+#else
       log_entry = i->second;
+#endif
     } else {
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+      std::unique_ptr<LogEntry> e(new LogEntry(path));
+      log_entry = e.get();
+      entries_.emplace(e->output, std::move(e));
```

Comment:
```suggestion
      std::string log_output = e->output;
      entries_.emplace(std::move(log_output), std::move(e));
```

In C++, `std::move(e)` might be evaluated before `e->output`.

Reply:
That's correct, parameter order evaluation is implementation defined.

Reply:
This is what I was unsure about... Thank you!

Reply:
This is what I was unsure about. Thank you!